src/runner_manager.py

```diff
@@ -290,6 +295,40 @@ def _get_runner_health_states(self) -> RunnerByHealth:
 
         return RunnerByHealth(healthy, unhealthy)
 
+    def _create_runner(
+        self, registration_token: str, resources: VirtualMachineResources, runner: Runner
+    ):
+        """Create a runner.
+
+        Issues RunnerInstalled metric if issue_metrics is set to True.
+
+        Args:
+            registration_token: Token for registering runner to GitHub.
+            resources: Configuration of the virtual machine resources.
+            runner: Runner to be created.
+        """
+        ts_now = time.time()
+        runner.create(
+            self.config.image,
+            resources,
+            RunnerManager.runner_bin_path,
+            registration_token,
+        )
+        ts_after = time.time()
+        if self.issue_metrics:
+            try:
+                metrics.issue_event(
+                    metrics.RunnerInstalled(
+                        timestamp=ts_after,
+                        flavor=self.app_name,
+                        duration=ts_after - ts_now,
+                    )
+                )
+            # We explicitly catch all exceptions here as we do not want to fail the
+            # runner creation process due to metrics issuing failure.
+            except Exception:  # pylint: disable=broad-except
```

Comment:
I'm not 100% convinced, this could be the sign of really big problems on the charm, in such cases that's better to let the exception go through and lead to error status

But if such critical issues occurs on the charm, it's likely that some other parts will also raise exception

Reply:
I narrowed the exception catch. My original thought was to obey the spec, which indicates that no error related to metrics should let the charm or runners stop working.